app/Http/Controllers/PostController.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Models\Post;
+
+class PostController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        //
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        //
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        //store the new post's information
+        $this->validate($request, [
+            'name' => 'required|max:20|min:1',
+            'description' => 'required|min:1|max:255',
+            'user_id' => 'required|integer'
+        ]);
+
+        $post = new Post();
+        $post->name = $request->name;
+        $post->description = $request->description;
+        $post->user_id = $request->user_id;
+        $result = $post->save();
+        if ($result) {
+            $message = 'Post successfully!';
+        } else {
+            $message = 'Post failed!';
+        }
```

Comment:
Laravel would throw an exception if it was unable to save, so this check is redundant.